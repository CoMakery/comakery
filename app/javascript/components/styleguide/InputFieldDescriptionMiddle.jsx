import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldDescriptionMiddle extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__description-middle',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="textarea"
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldDescriptionMiddle
