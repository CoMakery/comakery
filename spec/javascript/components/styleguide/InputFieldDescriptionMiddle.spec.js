import React from 'react'
import { shallow } from 'enzyme'
import InputFieldDescriptionMiddle from 'components/styleguide/InputFieldDescriptionMiddle'

describe('InputFieldDescriptionMiddle', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<InputFieldDescriptionMiddle />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.input-field__description-middle')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<InputFieldDescriptionMiddle className="__test" />)

    expect(wrapper.exists('.input-field__description-middle.__test')).toBe(true)
  })
})
