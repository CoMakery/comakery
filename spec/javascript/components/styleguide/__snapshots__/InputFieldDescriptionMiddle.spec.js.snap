// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldDescriptionMiddle renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__description-middle"
    copyOnClick={false}
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgInputRef={null}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    imgRequirements=""
    imgVerifier={[Function]}
    max=""
    min=""
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    recommended={false}
    required={false}
    selectEntries={Array []}
    step=""
    symbolLimit={100}
    title="title"
    type="textarea"
    value=""
  />
</Fragment>
`;
